src/database/adatabasesetup.cpp

```diff
@@ -328,13 +327,26 @@ bool ADataBaseSetup::importDefaultData()
         if (!query.exec()) {
             DEB << "Error: " << query.lastError().text();
         }
+
         //fill with data from csv
-        if (!commitData(aReadCsv(AStandardPaths::absPathOf(AStandardPaths::Templates)
-                                 % QLatin1Char('/')
-                                 % table % QStringLiteral(".csv")),
-                        table)) {
-            DEB << "Error importing data.";
-            return false;
+        if (use_local_data)
+        {
+            DEB << "Using local resources";
+            if (!commitData(aReadCsv(QStringLiteral(":templates/database/templates/")
+                                     + table + QStringLiteral(".csv")),
+                            table)) {
+                DEB << "Error importing data (local).";
+                return false;
+            }
+        } else {
+            DEB << "Using downloaded resources";
+            if (!commitData(aReadCsv(AStandardPaths::absPathOf(AStandardPaths::Templates)
+                                     % QLatin1Char('/')
+                                     % table % QStringLiteral(".csv")),
+                            table)) {
+                DEB << "Error importing data (remote).";
+                return false;
+            }
```

Comment:
This screams code duplications.
```c++
// if `/` is necessary then AStandardPaths should have it by default
// we should see how ".../directory/" is handled in general
// i assume Qt just ignores the last `/`

// Outside for loop
QString data_to_commit_path;
QString error_msg = "Error importing data ";
if(use_local_data){
	data_to_commit_path = ":templates/database/templates/";
	error_msg = "(local)";
} 
else {
	data_to_commit_path = AStandardPaths::absPathOf(AStandardPaths::Templates)
	error_msg = "(remote)";
}

// In for loop
if(!commitData(data_to_commit + table + ".csv", table)){
	DEB << error_msg;
	return false;
}
return true;
```
if it is expecting a file with name `table.csv` then why does `commitData` take it as a second parameter?

Reply:
#### Standard Paths
The strings provided by QStandardPaths do not include a trailing `/` character, as can be seen [here](https://doc.qt.io/qt-5/qstandardpaths.html#StandardLocation-enum). We need to add it when writing to a file, because otherwise what would happen is a file like this would be created:
`path/to/dirfile.ini`
instead of
`path/to/dir/file.ini`

I am not sure about the reason behind this, for me it would make sense to append a trailing `/` character to indicate we are referencing a directory, since without it, it may be ambiguous in some contexts. Adding the trailing separator to AStandardPaths might bring with it some complication with regard to cross-platform compatibility, QDir`s [toNativeSeperators](https://doc.qt.io/qt-5/qdir.html#toNativeSeparators) might be useful. Should we go for a solution like that we should document it clearly, since it is 'non-standard' behaviour for Qt - I'll look into it.
#### commitData / loop
`commitData()` takes the parsed data to be committed as a first argument, and the table it is supposed to commit it to as the second argument. These look like the same in this case, because the file is named like the table, so `aReadCsv()` and `commitData()` take the same argument, but that is not always the case.
I will rewrite it a little to make that distinction more visible, including you suggestions on the loop!

Reply:
Then we should rework AStandardPaths to be more robust. QStrings may be too generic and clunky for the specific use paths are. Since standard paths are directories (if we want standard files we can extend it of course) we could have them like this.
```c++
    dirs = {  // [G]: potential rename to `dirs`
        {Database, QDir(QStandardPaths::writableLocation(data_location))},
        {Templates, QDir(QStandardPaths::writableLocation(data_location) + "/templates")},
        {DatabaseBackup, QDir(QStandardPaths::writableLocation(data_location) + "/backup")}
    };
```
This seems like the simple way to construct the new path and i dont see any reason to overcomplicate it.
Making AStandardPaths QDirs instead of QString means that we can use the existing methods that deal with the plumbing of slashes and so on.
The above can be used as follows:
```c++
...
// more robust and safe since Qt deals with the path plumbing
if(!commitData(data_to_commit_dir.filePath(table + ".csv"), table)){ 
	DEB << error_msg;                                                                   
	return false;
}
```
In order to refactor `commitData` we can start by overloaing it first so that it deals with `QDir` logic and incrementaly delete the `QString` oveload if it is not needed.